### PR TITLE
Observer screen

### PR DESCRIPTION
**Pull request by BRHoward**
Added basic functionality for an observer screen.
Added a separate set of screens for the observer to see during the game, these screens currently just have basic info and which could be improved upon later.

### CODE REVIEW

**Comment by JamesAdamHughes** [comment]
plz rebase

File: client/Angular/directives/userProfileDirective.js [hunk 1]
```diff
@@ -43,15 +43,30 @@ ClonageApp.directive('userProfile', function() {
...
+					// get the image from the bot image url
+					fabric.Image.fromURL('../../includes/images/bot_icon.png', function(oImg) {
+						canvas.add(oImg);
+						scaleCanvas(scope.sizex, scope.sizey);
+
+					});
+				}
+			});
+
+// 			// if the image is a bot
```
Comment (JamesAdamHughes) [inline client/Angular/directives/userProfileDirective.js]: remove comments

File: client/Angular/services/roomService.js [hunk 2]
```diff
@@ -33,6 +33,20 @@ ClonageApp.service('roomService', ['communicationService', '$sessionStorage', fu
         return usersInRoom;
     }
 
+    function getActiveUsersInRoom() {
```
Comment (JamesAdamHughes) [inline client/Angular/services/roomService.js]: maybe comment this functionality?

File: server/BlackWhiteCards.json [hunk 3]
```diff
@@ -529,7 +529,7 @@
       "pick": 1
     },
     {
-      "text": "Tonight on the One Show, Joanna Lumley will be talking about _.",
+      "text": "Tonight on The One Show, Joanna Lumley will be talking about _.",
```
Comment (JamesAdamHughes) [inline server/BlackWhiteCards.json]: spelt "the One show", see http://ichef.bbci.co.uk/images/ic/480xn/p02wr8px.jpg

File: server/GameController.js [hunk 4]
```diff
@@ -564,12 +566,15 @@ module.exports = function(data) {
...
+		//updating the player ranks with the new point values
+		setRank();
+
 		//need also the send new point values back, doing this through playerRoundResults
 		var currentResults = rounds[rounds.length - 1].results;
 		callback(currentPlayer.hand, currentResults);
 	};
 
-	// TO DO : check game state before every move!
+	// TO DO : check game state before every move!`
```
Comment (JamesAdamHughes) [inline server/GameController.js]: remove quote symbol

File: server/GameController.js [hunk 5]
```diff
@@ -586,7 +591,7 @@ module.exports = function(data) {
 		currentAnswers.forEach(function(answer) {
 			answer.playersVote.forEach(function(votingPlayer) {
 				playersWhoHaventVoted = playersWhoHaventVoted.filter(function(iteratedPlayer) {
-					return (iteratedPlayer.name !== votingPlayer);
+					return (iteratedPlayer.name !== votingPlayer.name);
```
Comment (JamesAdamHughes) [inline server/GameController.js]: should check by id not name, incase of the same name

File: server/Player.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ function Player(user, cardController) {
 	var self = this;
 	self.uId = user.uId;
 	self.name = user.name;
+	self.image = user.image;
```
Comment (JamesAdamHughes) [inline server/Player.js]: why do we need the image in the player?

File: server/BlackWhiteCards.json (hunk 3 above)
Comment (BRHoward) [inline server/BlackWhiteCards.json]: https://en.wikipedia.org/wiki/The_One_Show

CTRL+F "The One Show"

File: server/server.js [hunk 7]
```diff
@@ -96,24 +96,25 @@ module.exports = function(port, enableLogging) {
             logger.debug("Final registered details of user are: " + user.name + " " + user.uId);
         });
 
-        /*
-            Set the user's name as given by the client
-         */
-        socket.on('USER set name', function(msg) {
-            user.name = msg.name;
+        socket.on('USER set profile', function(data) {
```
Comment (JamesAdamHughes) [inline server/server.js]: can these just call the same function, with isObserver parameterised?

**Comment by JamesAdamHughes** [comment]
happy

**Pull request merged**